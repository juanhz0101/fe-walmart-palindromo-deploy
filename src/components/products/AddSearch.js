import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

function AddSearch({ setQuery }) {

    const [inputValue, setInputValue] = useState('')

    useEffect(() => {
        setInputValue(inputValue);
        if (inputValue === ' ') {
            setInputValue('');
        }else{
            setQuery(q => ({ ...q, 'query': inputValue }))
        }
       
    }, [inputValue, setQuery]);

    return (
        
        <form>
            <input
                type="text" 
                value={inputValue}
                onChange={ (e) => setInputValue(e.target.value) }
                placeholder="Por favor ingrese un Nombre o ID de producto"
            />
        </form>
    )
}


AddSearch.propTypes = {
    setQuery: PropTypes.func.isRequired
}

export default AddSearch
