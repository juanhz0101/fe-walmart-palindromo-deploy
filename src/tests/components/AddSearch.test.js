import React from 'react';
import '@testing-library/jest-dom';
import { shallow} from 'enzyme';
import AddSearch from "../../components/products/AddSearch";


describe('Pruebas en <AddCategory />', () => {

    const setQuery = jest.fn();
    let wrapper = shallow( <AddSearch setQuery={setQuery} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddSearch setQuery={setQuery} />);
    })

    test('debe de mostrar <AddCategory /> correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    })
    
})