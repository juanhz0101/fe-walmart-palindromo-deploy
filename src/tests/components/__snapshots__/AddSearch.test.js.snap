// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrar <AddCategory /> correctamente 1`] = `
<form>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
